src/app/clusters/operational-state-server/operational-state-server.h

```diff
@@ -27,6 +27,55 @@ namespace app {
 namespace Clusters {
 namespace OperationalState {
 
+/**
+ * A class which represents the operational error event of an Operational State cluster derivation instance.
+ */
+class GenericErrorEvent : private app::Clusters::OperationalState::Events::OperationalError::Type
+{
+    using super = app::Clusters::OperationalState::Events::OperationalError::Type;
+
+public:
+    GenericErrorEvent(ClusterId aClusterId, const Structs::ErrorStateStruct::Type & aError) : mClusterId(aClusterId)
```

Comment:
This is actually a bit dangerous: if the ErrorStateStruct::Type goes out of scope before the GenericErrorEvent does, we will possibly end up with dangling pointers.

It would be safer to keep taking the ErrorStateStruct as an argument to OnOperationalErrorDetected and using the cluster id of the instance to construct the GenericErrorEvent from the struct and ensure that it does not live too long.  In particular, this event definition could move into operational-state-server.cpp, so it's not exposed.